app/views/course/discussion/_topic.json.jbuilder

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+json.createdAt topic.created_at
+json.read_marks read_marks if defined?(read_marks)
+
+posts = topic.posts
+posts = topics.posts.with_read_marks_for(current_user) if defined?(read_marks)
+posts = posts.calculated(:upvotes, :downvotes).with_user_votes(current_user) if defined?(with_votes)
+posts_locals = { posts: posts.ordered_topologically, max_depth: 3 }
+posts_locals[:post_partial] = post_partial if defined?(post_partial)
+posts_locals[:post_locals] = post_locals if defined?(post_locals)
+posts_locals[:max_depth] = max_depth if defined?(max_depth)
+
+json.postsLocal posts_locals
```

Comment:
Probably we dont need this right? in _topic.html.slim, I think post_locals is the variable being passed down recursively in html slim